addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -660,9 +660,21 @@ import wUtils from '@website/js/utils';
 
             switch (comparator) {
                 case 'contains':
-                    return value.includes(comparable);
                 case '!contains':
-                    return !value.includes(comparable);
+                    const isContains = comparator === 'contains';
+                    const isMultiValue = dependencyEl.type === 'checkbox' 
+                        || dependencyEl.type === 'radio'
+                        || dependencyEl.nodeName === 'SELECT';
+                    if (isMultiValue) {
+                        comparable = comparable.split(',');
+                        return isContains 
+                        ? value.some(v => comparable.includes(v))
```

Comment:
`comparable.includes(v)` why we need this ? It's cause unexpected behavior.

![image](https://github.com/user-attachments/assets/6c2e30e0-d40d-4fc5-a375-4d5fac35877b)

1. choose any field.
2. add visibility, `visibile only if` & select condition `contains`.
3. select the 4th option `Opt`.

-> field will be visible for all the option, as you are matching for substring here.

Reply:
we are not matching for substring here. we are searching for the string named `v` in the array of strings named `comparable` so i think above code will give expected result . I have tested it.

Reply:
Agree, thanks : )  I didn't tested in depth.